src/main/scala/wdltool/graph/package.scala

```diff
@@ -15,22 +15,19 @@ package object graph {
     }
 
     def graphName: String = dotSafe(graphNode match {
-      case c: CallNode =>
-        s"call ${c.name}"
-      case s: ScatterNode =>
-        s"scatter ${s.scatterVariableMapping.graphInputNode.name}"
-      case gin: GraphInputNode =>
-        s"${gin.womType.toWdlString} ${gin.name}"
-      case gon: GraphOutputNode =>
-        s"${gon.womType.toWdlString} ${gon.name}"
+      case c: CallNode => s"call ${c.name}"
+      case s: ScatterNode => s"scatter"
+      case gin: OptionalGraphInputNodeWithDefault => s"${gin.womType.toWdlString} ${gin.name} = ..."
+      case gin: GraphInputNode => s"${gin.womType.toWdlString} ${gin.name}"
+      case gon: GraphOutputNode => s"${gon.womType.toWdlString} ${gon.name}"
       case expr: ExpressionNode =>
         val inputNames = expr.instantiatedExpression.expression.inputs.mkString(", ")
         s"${expr.womType.toWdlString} ${expr.name} = f($inputNames)"
       case other =>
         throw new Exception(s"womgraph can't find a graphName for GraphNodes of type: ${other.getClass.getSimpleName}")
     })
 
-    def graphId: String = dotSafe("NODE" + graphNode.hashCode().toString)
+    def graphId: String = dotSafe("NODE" + System.identityHashCode(graphNode).toString)
```

Comment:
Can you refactor this into a `def graphNodeUniqueId: Int = System.identityHashCode(graphNode)`, with a comment that this **should** be good-enough, but on some JVMs could cause unwelcome behavior. https://stackoverflow.com/questions/10645494/can-i-assume-two-objects-with-the-same-system-identityhashcode-are-the-same#10645567

Reply:
😦